code/messaging/src/main/java/com/adobe/marketing/mobile/MessagingPushBuilder.java

```diff
@@ -0,0 +1,199 @@
+            final String channelId = DEFAULT_CHANNEL_ID;
+            if (notificationManager.getNotificationChannel(DEFAULT_CHANNEL_ID) != null) {
+                Log.debug(MessagingPushConstants.LOG_TAG, SELF_TAG, "Channel already exists for the default channel ID: " + channelId);
+                return DEFAULT_CHANNEL_ID;
+            } else {
+                Log.debug(MessagingPushConstants.LOG_TAG, SELF_TAG, "Creating a new channel for the default channel ID: " + channelId + ".");
+                final NotificationChannel channel = new NotificationChannel(channelId, DEFAULT_CHANNEL_NAME, NotificationManager.IMPORTANCE_DEFAULT);
+                notificationManager.createNotificationChannel(channel);
+            }
+            return channelId;
+        }
+    }
+
+
+    private void addActionButtons(final NotificationCompat.Builder builder) {
+        // TODO: Add this in the future Pull Request
+    }
+
+    /**
+     * Sets the small icon for the notification.
+     * If a small icon is received from the payload, the same is used.
+     * If a small icon is not received from the payload, we use the icon set using MobileCore.setSmallIcon().
+     * If a small icon is not set using MobileCore.setSmallIcon(), we use the default small icon of the application.
+     *
+     * @param builder the notification builder
+     */
+    private void setSmallIcon(final NotificationCompat.Builder builder) {
+        final int iconFromPayload = utils.getSmallIconWithResourceName(payload.getIcon());
+        final int iconFromMobileCore =  -1 ;   //MobileCore.getSmallIconResourceId();
+
+        if (isValidIcon(iconFromPayload)) {
+            builder.setSmallIcon(iconFromPayload);
+        } else if (isValidIcon(iconFromMobileCore)) {
+            builder.setSmallIcon(iconFromMobileCore);
+        } else {
+            final int iconFromApp = utils.getDefaultAppIcon();
+            if (isValidIcon(iconFromApp)) {
+                builder.setSmallIcon(iconFromApp);
+            } else {
+                Log.warning(MessagingPushConstants.LOG_TAG, SELF_TAG, "No valid small icon found. Notification will not be displayed.");
+            }
+        }
+    }
+
+    /**
+     * Sets the sound for the notification.
+     * If a sound is received from the payload, the same is used.
+     * If a sound is not received from the payload, the default sound is used
+     * The sound name from the payload should also include the format of the sound file. eg: sound.mp3
+     * @param notificationBuilder the notification builder
+     */
+    private void setSound(final NotificationCompat.Builder notificationBuilder) {
+        if(!StringUtils.isNullOrEmpty(payload.getSound())) {
+            notificationBuilder.setSound(utils.getSoundUriForResourceName(payload.getSound()));
+        }
+        notificationBuilder.setDefaults(Notification.DEFAULT_ALL);
```

Comment:
Looks like this API is deprecated in level 26. Does using this `notificationBuilder.setSound(RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION));` result in the same behavior?

Reply:
Yeap. `setDefaults` is deprecated. From API 26, sounds can only be set on the notification channel.
they have also deprecated the setSound API https://developer.android.com/reference/android/app/Notification.Builder#setSound(android.net.Uri)

Reply:
Let's keep.
notificationBuilder.setSound(RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION)); for setting sound for API 26 and below and make sure we document that fact that sound has to set on the Notification channel